### PR TITLE
Make the render cache work with all the data providers

### PR DESCRIPTION
**Pull request by e0ipso**
There are several `TODO` in the code base regarding this.

### CODE REVIEW

**Comment by e0ipso** [comment]
@amitaibu this is ready to be reviewed.

**Comment by e0ipso** [comment]
Now that I think of it, the cache invalidation part on update is still pending

**Comment by e0ipso** [comment]
@amitaibu tests are green, including new ones.

A semi-important bug was fixed on the render cache layer with this.

File: plugins/restful/RestfulBase.php [hunk 1]
```diff
@@ -1085,9 +1105,13 @@ protected function generateCacheId(array $context = array()) {
     $version = $this->getVersion();
     $cid = 'v' . $version['major'] . '.' . $version['minor'] . '::uu' . $this->getAccount()->uid . '::pa';
     $cid_params = array();
-    $request = $this->getRequest();
-    static::cleanRequest($request);
-    $options = $context + $request;
+    $options = $context;
+    if ($this->isReadMethod($this->getMethod())) {
+      // We don't want to split the cache with the body data on write requests.
```
Comment (amitaibu) [inline plugins/restful/RestfulBase.php:1110]: Can you explain what you mean by "split the cache"?

**Comment by amitaibu** [comment]
> A semi-important bug was fixed on the render cache layer with this.

Also can you clarify what was the bug :)

File: plugins/restful/RestfulBase.php (hunk 1 above)
Comment (e0ipso) [inline plugins/restful/RestfulBase.php:1110]: It means having different versions of the cache due to generating different cache ids.

This is the bug I was referring.

**Comment by amitaibu** [comment]
Merged, thanks.

**Pull request closed without merging**